src/m365/spo/commands/file/file-sharinglink-clear.ts

```diff
@@ -0,0 +1,175 @@
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const clearSharingLinks: () => Promise<void> = async (): Promise<void> => {
+      try {
+        if (this.verbose) {
+          logger.logToStderr(`Clearing sharing links for file ${args.options.fileUrl || args.options.fileId}${args.options.scope ? ` and scope ${args.options.scope}` : ''}`);
+        }
+
+        const fileDetails = await FileSharingLinkUtil.getFileDetails(args.options.webUrl, args.options.fileId, args.options.fileUrl);
+        const sharingLinks = await this.getFileSharingLinks(args.options.webUrl, logger, args.options.fileId, args.options.fileUrl, args.options.scope);
+        const batchRequests = sharingLinks.map((sharingLink, index) => {
+          return {
+            id: index,
+            method: 'DELETE',
+            url: `/sites/${fileDetails.SiteId}/drives/${fileDetails.VroomDriveID}/items/${fileDetails.VroomItemID}/permissions/${sharingLink.id}`
+          };
+        });
+        for (let i = 0; i < batchRequests.length; i += 20) {
+          const batchRequestChunk = batchRequests.slice(i, i + 20);
+          const requestOptions: CliRequestOptions = {
+            url: 'https://graph.microsoft.com/v1.0/$batch',
+            headers: {
+              accept: 'application/json',
+              'content-type': 'application/json'
+            },
+            responseType: 'json',
+            data: JSON.stringify({ requests: batchRequestChunk })
+          };
+          await request.post(requestOptions);
+        }
+      }
+      catch (err: any) {
+        this.handleRejectedODataJsonPromise(err);
+      }
+    };
+
+    if (args.options.confirm) {
+      await clearSharingLinks();
+    }
+    else {
+      const result = await Cli.prompt<{ continue: boolean }>({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to clear the sharing links of file ${args.options.fileUrl || args.options.fileId}${args.options.scope ? ` and scope ${args.options.scope}` : ''}?`
+      });
+
+      if (result.continue) {
+        await clearSharingLinks();
+      }
+    }
+  }
+
+  private async getFileSharingLinks(webUrl: string, logger: Logger, fileId?: string, fileUrl?: string, scope?: string): Promise<any[]> {
+    const sharingLinkListOptions: SpoFileSharingLinkListOptions = {
+      webUrl: webUrl,
+      fileId: fileId,
+      fileUrl: fileUrl,
+      scope: scope,
+      debug: this.debug,
+      verbose: this.verbose
+    };
+
+    const commandOutput = await Cli.executeCommandWithOutput(spoFileSharingLinkListCommand as Command, { options: { ...sharingLinkListOptions, _: [] } });
+    if (this.verbose) {
+      logger.logToStderr(commandOutput.stderr);
+    }
+
+    const outputParsed = JSON.parse(commandOutput.stdout);
+    if (outputParsed.length === 0) {
+      throw `There are no sharing links to be removed for the specific file.`;
```

Comment:
I'd say if there are not sharing links, we should just continue and don't throw an error.